lib/money/money/allocation.rb

```diff
@@ -2,22 +2,23 @@
 
 class Money
   class Allocation
-    # Splits a given amount in parts. The allocation is based on the parts' proportions
-    # or evenly if parts are numerically specified.
+    # Allocates a specified amount into parts based on their proportions or distributes
+    # it evenly when the number of parts is specified numerically.
     #
-    # The results should always add up to the original amount.
+    # The total of the allocated amounts will always equal the original amount.
     #
     # The parts can be specified as:
     #   Numeric — performs the split between a given number of parties evenly
     #   Array<Numeric> — allocates the amounts proportionally to the given array
     #
     # @param amount [Numeric] The total amount to be allocated.
     # @param parts [Numeric, Array<Numeric>] Number of parts to split into or an array (proportions for allocation)
-    # @param whole_amounts [Boolean] Specifies whether to allocate whole amounts only. Defaults to true.
+    # @param rounding_mode [Boolean, Integer] Specifies the rounding mode. If true, rounds to whole amounts.
+    #   If an integer, rounds to that decimal precision. Defaults to true (whole amounts).
     #
     # @return [Array<Numeric>] An array containing the allocated amounts.
     # @raise [ArgumentError] If parts is empty or not provided.
-    def self.generate(amount, parts, whole_amounts = true)
+    def self.generate(amount, parts, rounding_mode = true)
```

Comment:
I find that the name `rounding_mode` for this argument can be confused with the `Money.rounding_mode = BigDecimal::ROUND_HALF_EVEN` configuration.

Perhaps this should be called something like `decimal_cutoff` instead?